test/api_test.rb

```diff
@@ -71,7 +71,7 @@
       ensure
         tmp_input.close!
       end
-    end
+    end unless RUBY_ENGINE == 'truffleruby' # can't modify frozen String in truffleruby-19.0.0/lib/mri/delegate.rb:349:in `encode!'
```

Comment:
We don't force encoding in Asciidoctor anymore, so I'm curious where this call is coming from.

Reply:
Maybe it's a bug of `Tempfile.new` with the `:encoding` keyword, could you report it to https://github.com/oracle/truffleruby/issues ?

Reply:
@Mogztter?

Reply:
The full stack trace:

```console
Minitest::Test::TestAPI::TestLoad#test_should_not_load_file_with_unrecognized_encoding:
FrozenError: can't modify frozen String
    /home/guillaume/.rvm/rubies/truffleruby-19.1.0/lib/mri/delegate.rb:349:in `encode!'
    /home/guillaume/.rvm/rubies/truffleruby-19.1.0/lib/mri/delegate.rb:349:in `write'
    /home/guillaume/.rvm/rubies/truffleruby-19.1.0/lib/mri/delegate.rb:349:in `write'
    /home/guillaume/workspace/opensource/asciidoctor/asciidoctor/test/api_test.rb:64:in `test_should_not_load_file_with_unrecognized_encoding'
```

Reply:
@Mogztter  Could you report on TruffleRuby's bug tracker please?

Reply:
@eregon Yes, I will try to create a minimal reproduction case.